### PR TITLE
Update GTest

### PR DESCRIPTION
**Pull request by Knitschi**
* I added the new GTest version 1.8.0-hunter-p5.
* I modified the example to also use the GMock library.
* Increased the default GTest version to the new package version.

### CODE REVIEW

File: examples/GTest/CMakeLists.txt [hunk 1]
```diff
@@ -6,14 +6,16 @@ cmake_minimum_required(VERSION 3.0)
...
-
 project(download-gtest)
 
 # download gtest
 hunter_add_package(GTest)
 
-# now gtest can be used
-find_package(GTest CONFIG REQUIRED)
+# Use package GTest if only GTest is required
+# find_package(GTest CONFIG REQUIRED)
```
Comment (ruslo) [inline examples/GTest/CMakeLists.txt]: Please add a new test. Old code should work too.

**Comment by Knitschi** [comment]
I added another target that only uses the GTest library.

File: examples/GTest/foo.cpp [hunk 2]
```diff
@@ -1,5 +0,0 @@
-#include <gtest/gtest.h>
```
Comment (ruslo) [inline examples/GTest/foo.cpp]: Please leave this example untouched. This is a quite common pattern to use GTest and it need to be tested. Add separate example for other cases, like `examples/GTest-no-main`, `examples/GMock`.

Comment (Knitschi) [inline examples/GTest/foo.cpp]: Ok I just read that the main function is not needed when linking to GTest::main. I will write something about that in the comments and change the content of useGTest.cpp back to the former content of foo.cpp

Comment (Knitschi) [inline examples/GTest/foo.cpp]: BTW does the executable need to be named foo for your automated tests?

Comment (ruslo) [inline examples/GTest/foo.cpp]: > BTW does the executable need to be named foo for your automated tests?

No you can name it as you want. CI tests do check only build/link stage.

File: examples/GTest/CMakeLists.txt [hunk 3]
```diff
@@ -6,14 +6,23 @@ cmake_minimum_required(VERSION 3.0)
...
+# Create test executable that only uses GTest.
 find_package(GTest CONFIG REQUIRED)
+add_executable(GTestExe useGTest.cpp)
+target_link_libraries(GTestExe GTest::main)
 
-add_executable(foo foo.cpp)
-target_link_libraries(foo GTest::main)
+# Create test executable uses GMock and a custom main function.
+# Note that GMock also includes GTest so we do not need to link to GTest explicitly
+find_package(GMock CONFIG REQUIRED)
```
Comment (ruslo) [inline examples/GTest/CMakeLists.txt]: Please leave the original example as is. Create separate `examples/GMock/CMakeLists.txt` example for GMock testing. Thanks.

**Comment by Knitschi** [comment]
Ok I reverted all changes to the examples/GTest folder.

**Comment by ruslo** [comment]
> Ok I reverted all changes to the examples/GTest folder.

So now no GMock example at all... okay.

Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1139
* https://travis-ci.org/ingenue/hunter/builds/170985773

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1140
* https://travis-ci.org/ingenue/hunter/builds/170991090

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.16.18

**Pull request merged**